File: ELibrary.Standard/Objects/ELong.cs
using System;
using System.Drawing;

namespace ELibrary.Standard.Objects
{
    public class ELong
    {
        /// <summary>
        /// Converts a string to a long.
        /// </summary>
        /// <param name="obj">The string to convert.</param>
        /// <returns>The long value of the string or 0 if conversion fails.</returns>
        public static long valueOf(string obj)
        {
            if (string.IsNullOrWhiteSpace(obj))
                return 0L;

            try
            {
                return Convert.ToInt64(obj.Trim());
            }
            catch
            {
                return 0L;
            }
        }

        /// <summary>
        /// Converts a DBNull to a long (always 0).
        /// </summary>
        public static long valueOf(DBNull obj)
        {
            return 0L;
        }

        /// <summary>
        /// Converts a double to a long by rounding.
        /// </summary>
        public static long valueOf(double obj)
        {
            return (long)Math.Round(obj);
        }

        /// <summary>
        /// Converts a decimal to a long by rounding.
        /// </summary>
        public static long valueOf(decimal obj)
        {
            return (long)Math.Round(obj);
        }

        /// <summary>
        /// Converts a boolean to a long (true = 1, false = 0).
        /// </summary>
        public static long valueOf(bool obj)
        {
            return obj ? 1L : 0L;
        }

        /// <summary>
        /// Converts a Color to a long based on its ARGB value.
        /// </summary>
        public static long valueOf(Color obj)
        {
            return obj.ToArgb();
        }

        /// <summary>
        /// Converts a short to a long.
        /// </summary>
        public static long valueOf(short obj)
        {
            return obj;
        }

        /// <summary>
        /// Converts an int to a long.
        /// </summary>
        public static long valueOf(int obj)
        {
            return obj;
        }

        /// <summary>
        /// Returns the long value directly.
        /// </summary>
        public static long valueOf(long obj)
        {
            return obj;
        }

        /// <summary>
        /// Converts an object to a long by determining its type.
        /// </summary>
        public static long valueOf(object obj)
        {
            if (obj == null || obj is DBNull)
                return 0L;

            try
            {
                switch (obj)
                {
                    case string strValue:
                        return valueOf(strValue);

                    case double doubleValue:
                        return valueOf(doubleValue);

                    case decimal decimalValue:
                        return valueOf(decimalValue);

                    case int intValue:
                        return intValue;

                    case long longValue:
                        return longValue;

                    case short shortValue:
                        return shortValue;

                    case bool boolValue:
                        return valueOf(boolValue);

                    case Color colorValue:
                        return valueOf(colorValue);

                    default:
                        return 0L; // Unconvertible type
                }
            }
            catch
            {
                return 0L;
            }
        }
    }
}

File: ELibrary.Standard.VB/Objects/ELong.vb
Imports System.Drawing
Imports ELibrary.Standard.VB.Modules

Namespace Objects

    Public Class ELong

        ''' <summary>
        ''' Converts An Object to long
        ''' </summary>
        ''' <param name="obj"></param>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Public Overloads Shared Function ValueOf(ByVal obj As String) As Long
            Try

                If basExtensions.IsNothing(obj) Then Return 0
                If obj = "" Then Return 0
                If obj.Trim = "" Then Return 0

                Return CLng(obj)

            Catch ex As Exception
                Return 0

            End Try

        End Function


        ''' <summary>
        ''' Converts An Object to long
        ''' </summary>
        ''' <param name="obj"></param>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Public Overloads Shared Function ValueOf(ByVal obj As System.DBNull) As Long
            Try

                Return 0

            Catch ex As Exception
                Return 0

            End Try

        End Function

        ''' <summary>
        ''' Converts An Object to long
        ''' </summary>
        ''' <param name="obj"></param>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Public Overloads Shared Function ValueOf(ByVal obj As Double) As Long
            Try

                Return CLng(obj)

            Catch ex As Exception
                Return 0

            End Try

        End Function


        ''' <summary>
        ''' Converts An Object to long
        ''' </summary>
        ''' <param name="obj"></param>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Public Overloads Shared Function ValueOf(ByVal obj As Decimal) As Long
            Try

                Return CLng(obj)

            Catch ex As Exception
                Return 0

            End Try

        End Function

        Public Overloads Shared Function ValueOf(ByVal obj As Boolean) As Long
            Try

                Return Math.Abs(CLng(obj))

            Catch ex As Exception
                Return 0

            End Try

        End Function


        Public Overloads Shared Function ValueOf(ByVal obj As Color) As Long
            Try

                Return obj.ToArgb

            Catch ex As Exception
                Return 0

            End Try

        End Function

        Public Overloads Shared Function ValueOf(ByVal obj As Int16) As Long
            Return obj
        End Function

        Public Overloads Shared Function ValueOf(ByVal obj As Int32) As Long
            Return obj
        End Function

        Public Overloads Shared Function ValueOf(ByVal obj As Int64) As Long
            Return obj
        End Function

        Public Overloads Shared Function ValueOf(ByVal obj As Object) As Long
            If obj Is Nothing Then Return 0

            If TypeOf obj Is String Then Return ValueOf(CType(obj, String))
            If TypeOf obj Is DBNull Then Return 0
            If TypeOf obj Is Double Then Return ValueOf(CType(obj, Double))
            If TypeOf obj Is Long Then Return CLng(obj)
            If TypeOf obj Is Int32 Then Return ValueOf(CType(obj, Int32))
            If TypeOf obj Is Decimal Then Return ValueOf(CType(obj, Decimal))
            If TypeOf obj Is Int16 Then Return ValueOf(CType(obj, Int16))
            If TypeOf obj Is Boolean Then Return ValueOf(CType(obj, Boolean))
            If TypeOf obj Is Color Then Return ValueOf(CType(obj, Color))

            Return 0    REM Cant convert this
        End Function


    End Class

End Namespace